### ISSUE TITLE
Allow wrapping function like `mem` does

### ISSUE THREAD

**Issue by fregante** [sindresorhus/p-retry#61]
Instead of passing a function to immediately call it, I'd like to wrap a function to be used later. From:

```js
function runMeNow() {}
function init() {
	pRetry(runMeNow, {
		bigOptionsObject
	}).then(console.log)
}

init();
```

to:

```js
const runMeRetried = pRetry.wrap(runMeNow, {
	bigOptionsObject
});
function init() {
	runMeRetried.then(console.log)
}

init();
```

**Comment by sindresorhus** [comment]
Sure. It should probably be a named export though. Naming suggestions?

- `makeRetriable`
- `createRetry`

**Comment by jrobeson** [comment]
I was looking for something to handle retries and i found this and node-retry.  node-retry has `retry.wrap` https://github.com/tim-kos/node-retry#retrywrapobj-options-methodnames  which can wrap an entire object or just specific methods.

I ended up going with p-retry due to ts support, but `wrap` seemed very useful for a particular object I have.

**Comment by fregante** [comment]
Yes that's it. You make it look simple 😅 the only bothersome part that prevents me from doing exactly that would be the types

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/p-retry/releases/tag/v7.0.0-0
